packages/scheduler/lib/models/tasks.ts

```diff
@@ -0,0 +1,246 @@
+        .where('id', taskId)
+        .update({
+            state: transition.value.to,
+            last_state_transition_at: new Date(),
+            terminated: terminalStates.includes(transition.value.to),
+            output: output || null
+        })
+        .returning('*');
+    if (!updated?.[0]) {
+        return Err(new Error(`Task with id '${taskId}' not found`));
+    }
+    return Ok(DbTask.from(updated[0]));
+}
+
+export async function dequeue({ groupKey, limit }: { groupKey: string; limit: number }): Promise<Result<Task[]>> {
+    try {
+        const tasks = await db
+            .update({
+                state: 'STARTED',
+                last_state_transition_at: new Date()
+            })
+            .from<DbTask>(TASKS_TABLE)
+            .whereIn(
+                'id',
+                db
+                    .select('id')
+                    .from<DbTask>(TASKS_TABLE)
+                    .where({ group_key: groupKey, state: 'CREATED' })
+                    .where('starts_after', '<=', db.fn.now())
+                    .orderBy('created_at')
+                    .limit(limit)
+                    .forUpdate()
+                    .skipLocked()
+            )
+            .returning('*');
+        if (!tasks?.[0]) {
+            return Ok([]);
+        }
+        // Sort tasks by id (uuidv7) to ensure ordering by creation date
+        const sorted = tasks.sort((a, b) => a.id.localeCompare(b.id)).map(DbTask.from);
+        return Ok(sorted);
+    } catch (err: unknown) {
+        return Err(new Error(`Error dequeuing tasks for group key '${groupKey}': ${stringifyError(err)}`));
+    }
+}
+
+export async function expiresIfTimeout(): Promise<Result<Task[]>> {
+    try {
+        const tasks = await db
+            .update({
+                state: 'EXPIRED',
+                last_state_transition_at: new Date(),
+                terminated: true
+            })
+            .from<DbTask>(TASKS_TABLE)
+            .whereIn(
+                'id',
+                db
+                    .select('id')
+                    .from<DbTask>(TASKS_TABLE)
+                    .where((builder) => {
+                        builder
+                            .where({ state: 'CREATED' })
+                            .andWhere(db.raw(`starts_after + created_to_started_timeout_secs * INTERVAL '1 seconds' < CURRENT_TIMESTAMP`));
+                        builder
+                            .orWhere({ state: 'STARTED' })
+                            .andWhere(db.raw(`last_heartbeat_at + heartbeat_timeout_secs * INTERVAL '1 seconds' < CURRENT_TIMESTAMP`));
+                        builder
+                            .orWhere({ state: 'STARTED' })
+                            .andWhere(db.raw(`last_state_transition_at + started_to_completed_timeout_secs * INTERVAL '1 seconds' < CURRENT_TIMESTAMP`));
+                    })
+                    .forUpdate()
+                    .skipLocked()
+                    .debug(true)
```

Comment:
Intentional I presume to leave this in while under development @TBonnin ?

Reply:
no actually it is some dev leftover. I will remove it